var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -232,6 +232,12 @@ class Gromacs(CMakePackage, CudaPackage):
         for gmx_ver, plumed_vers in plumed_patches.items():
             depends_on("plumed@{0}".format(plumed_vers), when="@{0}+plumed".format(gmx_ver))
 
+    variant(
+        "spack_intel",
+        default=False,
```

Comment:
I agree that this would be ideal, but I don't see a good way of finding that the intel compiler has been installed through spack. Even if there is a package `intel-oneapi-compiler`, the `intel` compiler use could still be from an external source.